typings/index.d.ts

```diff
@@ -2889,19 +2889,56 @@ export interface ApplicationCommandData {
   defaultPermission?: boolean;
 }
 
-export interface ApplicationCommandOptionData {
-  type: ApplicationCommandOptionType | ApplicationCommandOptionTypes;
+export type CommandOptionDataTypeResolvable = ApplicationCommandOptionType | ApplicationCommandOptionTypes;
+
+export type CommandOptionChoiceResolvableType =
+  | ApplicationCommandOptionTypes.NUMBER
+  | 'NUMBER'
+  | ApplicationCommandOptionTypes.STRING
+  | 'STRING'
+  | ApplicationCommandOptionTypes.INTEGER
+  | 'INTEGER';
+
+export type CommandOptionSubOptionResolvableType =
+  | ApplicationCommandOptionTypes.SUB_COMMAND
+  | 'SUB_COMMAND'
+  | ApplicationCommandOptionTypes.SUB_COMMAND_GROUP
+  | 'SUB_COMMAND_GROUP';
+
+export type CommandOptionNonChoiceResolvableType = Exclude<
+  CommandOptionDataTypeResolvable,
+  CommandOptionChoiceResolvableType | CommandOptionSubOptionResolvableType
+>;
+
+export interface BaseApplicationCommandOptionsData {
   name: string;
   description: string;
   required?: boolean;
+}
+
+export interface ApplicationCommandChoicesData extends BaseApplicationCommandOptionsData {
+  type: CommandOptionChoiceResolvableType;
   choices?: ApplicationCommandOptionChoice[];
-  options?: ApplicationCommandOptionData[];
 }
 
-export interface ApplicationCommandOption extends ApplicationCommandOptionData {
+export interface ApplicationCommandSubOptionsData extends BaseApplicationCommandOptionsData {
+  type: CommandOptionSubOptionResolvableType;
+  options?: this[];
```

Comment:
This is inaccurate. In the case of a subcommand group, options is just an "array" of subcommands (usually always has just one value anyways)
In the case of a subcommand, options is any valid option type, excluding subcommand/subcommand group

Reply:
Good catch, this has been fixed.